dali/sasha/saarch.cpp

```diff
@@ -85,295 +85,444 @@ static void mkDateCompare(bool dfu,const char *dt,StringBuffer &out,char fill)
+
+                        if (!addOutput(outputFields, outputModifiedTime, hasWUSOutput, wuTree, dt, WUSbuf))
+                        {
+                            overflowed = true;
                             break;
+                        }
                     }
+                    wuCount++;
+                    if (wuCount == numWUs)
+                        break;
                 }
-                if (index>start+num)
+                if (overflowed || (wuCount == numWUs))
                     break;
             }
+            if (sortInc)
+            {
+                ForEachItemInRev(i, wus)
+                {
+                    CWUData &wuData = wus.item(i);
+                    if (!addOutput(outputFields, outputModifiedTime, hasWUSOutput, wuData.wuTree, wuData.modifiedTime, WUSbuf))  //Log an error?
+                        break;
+                }
+            }
+
+            if (hasWUSOutput)
+                cmd->setWUSresult(WUSbuf);
         }
-    }
-    if (haswusoutput)
-        cmd->setWUSresult(WUSbuf);
+        void getOnlineWUs()
+        { //The code here is refactered based on the pre-existing code.
+            if (cmd->getAction()==SCA_WORKUNIT_SERVICES_GET)
+                throw MakeStringException(-1,"SCA_WORKUNIT_SERVICES_GET not implemented for online workunits!");
+
+            Owned<IRemoteConnection> conn;
+            setSDSConnection(conn, mask);
+            if (!conn)
+                return;
+    
+            CIArrayOf<CWUData> wus;
+            unsigned wuCount = 0;
+            StringBuffer wuid;
+            Owned<IPropertyTreeIterator> iter = conn->queryRoot()->getElements(isWild ? "*" : NULL);
+            ForEach(*iter)
+            {
+                IPropertyTree *pt = &iter->query();
+                wuid.set(pt->queryName());
+                if (wuid.isEmpty())
+                    continue;
+
+                if (!parseAndCheckWUID(wuid, true))
+                    continue;
+
+                try
+                {
+                    if (!checkFilters(pt, true))
+                        continue;
+                }
+                catch (IException *e)
+                {
+                    VStringBuffer msg("WUiterate: Workunit %s failed to load", wuid.str());
+                    EXCLOG(e,msg.str());
+                    e->Release();
+                    continue;
+                }
+
+                if (sortInc)
+                {//The output should contain WUs from new to old. So, we need to reverse the order before output.
+                    Owned<CWUData> wu = new CWUData();
+                    wu->wuTree.setown(createPTreeFromIPT(pt));
+                    wus.append(*wu.getClear());
+                }
+                else if (!addOutputFromWUTree(outputFields, pt))
+                    break;
+
+                wuCount++;
+                if (wuCount == numWUs)
+                    break;
+            }
+            if (sortInc)
```

Comment:
see comment above, I don't think your elements are sorted, so reversing won't help. You will instead need to actually sort them.

Reply:
Will add a sort function.